bftengine/src/preprocessor/PreProcessor.cpp

```diff
@@ -928,9 +952,14 @@ void PreProcessor::onMessage<PreProcessRequestMsg>(PreProcessRequestMsg *message
   const string reqType = msg->reqType() == REQ_TYPE_PRE_PROCESS ? "REQ_TYPE_PRE_PROCESS" : "REQ_TYPE_CANCEL";
   LOG_DEBUG(logger(),
             "Received PreProcessRequestMsg" << KVLOG(
-
                 reqType, msg->reqSeqNum(), msg->getCid(), msg->senderId(), msg->clientId(), msg->reqOffsetInBatch()));
-  handleSinglePreProcessRequestMsg(msg);
+  if (msg->viewNum() != myReplica_.getCurrentView()) {
+    LOG_INFO(logger(),
+             "Ignore PreProcessRequestMsg as the view is invalid" << KVLOG(
+                 msg->reqSeqNum(), msg->senderId(), msg->clientId(), msg->reqOffsetInBatch(), message->viewNum()));
```

Comment:
In order to identify a request, we need to print in each log clientId, seqNum, and cid. Please ann cid.